### ISSUE TITLE
error unmarshalling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type releaseutil.SimpleHead

### ISSUE THREAD

**Issue by nrodruck** [helm/helm#10146]
I'm trying to port a simple text file into a configmap using the yaml file below. If the text file is empty the `dry-run` will complete without issue. If I place any content into the text file, or any other file I'm experimenting with (I've also used xml and yaml) I get the following error:

> Error: YAML parse error on chart_configmap/templates/test.txt: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type releasetutil.SimpleHead

**Folder Structure**

![image](https://user-images.githubusercontent.com/83596641/133672418-e21dc036-1d33-4603-95c6-9aeb3d1dcad1.png)

**ConfigMap**

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: helm-configmap
data:
  {{ (.Files.Glob "test.txt").AsConfig | indent 2 }}
```

Output of `helm version`:

![image](https://user-images.githubusercontent.com/83596641/133666710-a1ec6ba7-475a-4ee4-86a0-5335f9dfdbc0.png)

Output of `kubectl version`:

![image](https://user-images.githubusercontent.com/83596641/133666824-4cffd4e9-e35f-47aa-8c5b-abefa60ce099.png)

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

OpenShift

![image](https://user-images.githubusercontent.com/83596641/133672287-5e983c8d-c156-4e4e-a76e-2d59baa32d7b.png)

**Comment by yxxhero** [comment]
@nrodruck Please show your `test.txt` file. Thanks very much.

**Comment by zonggen** [comment]
I think this is duplicate to https://github.com/helm/helm/issues/5550 and an improvement (not direct fix) was proposed in https://github.com/helm/helm/pull/9941.

**Comment by bacongobbler** [comment]
closing as a duplicate.

**Comment by koosala** [comment]
ok, I ran into this same error and I made the same error as above. 

Note - you have added the `test.txt` file within the templates folder, and helm will try to process any file within the template folder as `yml` / `json` leading to the error.

Just keep the text file outside template and it can be referred as done in the script and things should work just fine. 

Running `helm install` with ` --dry-run --debug` will give more info.
